src/cdk-experimental/popover-edit/table-directives.ts

```diff
@@ -179,6 +209,40 @@ export class CdkPopoverEdit<C> implements AfterViewInit, OnDestroy {
         this.viewContainerRef,
         {$implicit: this.context}));
     this.focusTrap!.focusInitialElementWhenReady();
+
+    // Update the size of the popup initially and on subsequent changes to
+    // scroll position and viewport size.
+    merge(this.scrollDispatcher.scrolled(), this.viewportRuler.change())
+        .pipe(
+            startWith(null),
+            takeUntil(this.overlayRef!.detachments()),
+            )
+        .subscribe(() => {
+          this._updateOverlaySize();
+        });
+  }
+
+  private _getOverlayCells(): HTMLElement[] {
+    const cell = closest(this.elementRef.nativeElement!, CELL_SELECTOR) as HTMLElement;
+
+    if (!this._colspan.before && !this._colspan.after) {
+      return [cell];
+    }
+
+    const row = closest(this.elementRef.nativeElement!, ROW_SELECTOR)!;
+    const rowCells = Array.from(row.querySelectorAll(CELL_SELECTOR)) as HTMLElement[];
+    const ownIndex = rowCells.indexOf(cell);
```

Comment:
Since you're only using the array for the `indexOf`, you could shorten it a bit to `Array.prototype.indexOf.call(row.querySelectorAll(CELL_SELECTOR), cell)`.